frame/nicks/src/lib.rs

```diff
@@ -45,88 +45,85 @@ use sp_std::prelude::*;
 use sp_runtime::{
 	traits::{StaticLookup, Zero}
 };
-use frame_support::{
-	decl_module, decl_event, decl_storage, ensure, decl_error,
-	traits::{Currency, EnsureOrigin, ReservableCurrency, OnUnbalanced, Get},
-};
-use frame_system::ensure_signed;
+use frame_support::traits::{Currency, ReservableCurrency, OnUnbalanced};
+pub use pallet::*;
 
 type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
 type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
 
-pub trait Config: frame_system::Config {
-	/// The overarching event type.
-	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_system::{ensure_signed, pallet_prelude::*};
+	use frame_support::{ensure, pallet_prelude::*, traits::{EnsureOrigin, Get}};
+	use super::*;
 
-	/// The currency trait.
-	type Currency: ReservableCurrency<Self::AccountId>;
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
-	/// Reservation fee.
-	type ReservationFee: Get<BalanceOf<Self>>;
+		/// The currency trait.
+		type Currency: ReservableCurrency<Self::AccountId>;
 
-	/// What to do with slashed funds.
-	type Slashed: OnUnbalanced<NegativeImbalanceOf<Self>>;
+		/// Reservation fee.
+		#[pallet::constant]
+		type ReservationFee: Get<BalanceOf<Self>>;
 
-	/// The origin which may forcibly set or remove a name. Root can always do this.
-	type ForceOrigin: EnsureOrigin<Self::Origin>;
+		/// What to do with slashed funds.
+		type Slashed: OnUnbalanced<NegativeImbalanceOf<Self>>;
 
-	/// The minimum length a name may be.
-	type MinLength: Get<usize>;
+		/// The origin which may forcibly set or remove a name. Root can always do this.
+		type ForceOrigin: EnsureOrigin<Self::Origin>;
 
-	/// The maximum length a name may be.
-	type MaxLength: Get<usize>;
-}
+		/// The minimum length a name may be.
+		#[pallet::constant]
+		type MinLength: Get<u32>;
```

Comment:
I guess it should be fine to change it to `Get<usize>` here? The type in metadata is `Get<u32>`. Cannot find a better way to keep metadata type as `Get<u32>` type. `#[pallet::extra_constant]` seems for this scenario but the constant name will change, as functions in it should be named as snake case.

Reply:
yes it is better now to me too